File: tutu-backend/src/main/scala/fi/oph/tutu/backend/domain/HakemusListItem.scala
package fi.oph.tutu.backend.domain

import java.time.LocalDateTime

case class HakemusListItem(
  hakija: String,
  saapumisPvm: Option[LocalDateTime],
  hakemusOid: String,
  hakemusKoskee: Int,
  esittelijaOid: Option[String] = None,
  asiatunnus: Option[String] = None,
  esittelijaKutsumanimi: String,
  esittelijaSukunimi: String,
  kasittelyVaihe: KasittelyVaihe,
  muokattu: Option[LocalDateTime] = None,
  taydennyspyyntoLahetetty: Option[LocalDateTime] = None,
  ataruHakemustaMuokattu: Option[LocalDateTime] = None,
  apHakemus: Option[Boolean] = None,
  viimeinenAsiakirjaHakijalta: Option[LocalDateTime] = None,
  onkoPeruutettu: Option[Boolean] = None
)

File: tutu-frontend/src/lib/types/hakemusListItem.ts
import { HakemusKoskee } from './hakemus';
import { HakijaListItem } from './hakija';

export type HakemusListItem = {
  asiatunnus: string;
  hakija: HakijaListItem;
  saapumisPvm: string;
  hakemusOid: string;
  hakemusKoskee: HakemusKoskee;
  esittelijaOid: string | null;
  esittelijaKutsumanimi: string | null;
  esittelijaSukunimi: string | null;
  kasittelyVaihe: string;
  muokattu: string | null;
  taydennyspyyntoLahetetty?: string | null;
  ataruHakemustaMuokattu?: string | null;
  apHakemus?: boolean | null;
  viimeinenAsiakirjaHakijalta?: string | null;
  onkoPeruutettu?: boolean | null;
};
